src/NuGet.Core/NuGet.ProjectModel/LockFileDependencyProvider.cs

```diff
@@ -38,10 +38,9 @@ public LockFileDependencyProvider(LockFile lockFile)
             }
         }
 
-        public bool SupportsType(string libraryType)
+        public bool SupportsType(LibraryTypeFlag libraryType)
         {
-            return string.IsNullOrEmpty(libraryType) ||
-                   string.Equals(libraryType, LibraryTypes.Package);
+            return (libraryType & LibraryTypeFlag.Package) == LibraryTypeFlag.Package;
```

Comment:
I'm OK with that in principle but `HasFlag` has notoriously awful performance due to all the extra type and bounds checking it does.